app/models/mixins/miq_provision_quota_mixin.rb

```diff
@@ -375,7 +375,13 @@ def storage(prov, cloud, vendor, flavor_obj = nil)
       return nil unless flavor_obj
       flavor_obj.root_disk_size.to_i + flavor_obj.ephemeral_disk_size.to_i + flavor_obj.swap_disk_size.to_i
     else
-      prov.kind_of?(MiqRequest) ? prov.vm_template.provisioned_storage : prov.miq_request.vm_template.provisioned_storage
+      infra_storage(prov.kind_of?(MiqRequest) ? prov : prov.miq_request).to_i
+    end
+  end
+
+  def infra_storage(request)
+    if request.try(:vm_template).present?
```

Comment:
Looking into this more `vm_template` is just an alias for `source` [app/models/miq_provision_request.rb#L2](https://github.com/ManageIQ/manageiq/blob/fbdd937efefcbc407d32136183417b95de761f97/app/models/miq_provision_request.rb#L2) which is the same problem we have seen on Service Requests when the Service Template has been removed.

It would be better if we could remove all requests that do not have a valid `source` anymore instead of dealing with it at this lower level.